python/tank_vendor/shotgun_authentication/user_impl.py

```diff
@@ -276,6 +300,19 @@ def create_sg_connection(self):
             http_proxy=self._http_proxy,
         )
 
+    def are_credentials_valid(self):
+        """
+        Checks if the credentials for the user are valid.
+
+        :returns: True if the credentials are valid, False otherwise.
+        """
+        sg = self.create_sg_connection()
```

Comment:
This may be a silly comment, but wondering if we could just return True in this case? Basically shifting the responsibility from the API to the user to ensure a valid set of credentials in the case of a script user being used. I guess I am just thinking about performance... Feel free to ignore this if the comment is totally off :)